### ISSUE TITLE
Determine OSBUILD at end of build to handle security updates changing build number

### ISSUE THREAD

**Issue by n8felton** [MagerValp/AutoDMG#162]
When updating a base 10.11.6 (15G31) image from the MAS with the Security Update 2016-002 for El Cap, the resulting image would be build `15G1108`, but the `%OSBUILD%` variable (and resulting filename) is still `15G31`.

The thought is to grab the OSBUILD number between the install phase and the asr phase.

```
Script phase: install 3:/Users/admin/Library/Application Support/AutoDMG/Updates/Safari10.0.1ElCapitan(d1adda8).pkg
Starting phase: Installing Safari 10.0.1 for El Capitan
phase: Installing Safari 10.0.1 for El Capitan
Script phase: install 4:/private/tmp/dmg.uGdRrT/SecUpd2016-002ElCapitan.pkg
Starting phase: Installing Security Update 2016-002 El Capitan
phase: Installing Security Update 2016-002 El Capitan

## maybe here?

Script phase: asr
Starting phase: Converting disk image
phase: Converting disk image
```
Conversation at https://macadmins.slack.com/archives/autodmg/p1478044259001783

**Comment by MagerValp** [comment]
It's not entirely clear cut that the filename should match the About dialog, as it indicates which installer was used to build the image, not necessarily the result after patches. But I'll agree that it's confusing and inelegant.

I've been looking at implementing this, and it's complicated by the fact that all [the logic that handles the output path](https://github.com/MagerValp/AutoDMG/blob/673f7f982bc1a33b1b5f64446ac0fe83f6058361/AutoDMG/IEDCLIController.py#L186-L204) happens before the workflow even starts. Since the build should fail if the output path exists and the force flag wasn't specified, it'd be pretty frustrating if it failed at the very last step.

Another approach would be to put information about it in the UpdateProfiles, but that adds to the maintenance burden, which I'd like to avoid.

**Comment by arubdesu** [comment]
My knee-jerk reaction is 'not worth it', but that may just be a 'change-adverse' response - the filename means the base it was built off of to me, and possibly many others. If as little effort as postflighting a check of the build number contained within the image could be accomplished it would be different, but I'm really less concerned about the filename in the grand scheme of things - it's always wrapped in a workflow that tells me which one is valid/QA'd/approved for usage in a given scenario.

**Comment by nmcspadden** [comment]
I'm inclined to agree that the source is the more definitive and useful number, but perhaps simply display a warning if the build number changes after any package install? That way the admin can grep the logs and make decisions about what to do afterwards manually, with sufficient information on how to make a useful decision, without actually having to change how the build operates.

**Comment by n8felton** [comment]
I don't see any real added benefit to having the filename match the final image build if it's not an easy change. As you said, it can just be slightly confusing and is inelegant, but it's not necessarily a show stopper, just a minor inconsistency.

I do like @nmcspadden's suggestion of at least logging/notifying the admin that the build has changed. Something like 
```
phase: Installing Security Update 2016-002 El Capitan
Note: Security Update 2016-002 El Capitan has updated the image's OS build number to 15G1108
```

**Comment by MagerValp** [comment]
Unconditionally logging the final build number after the image is done would be trivial, so that's a given. Notifying of change could probably be done from within `installesdtodmg.sh` without involving the upper layers at all, I'll see if I can whip something up.

Sleeping on it helped, and I think I can make a reasonably clean implementation of automatic renaming, but it'd still be a lot of work. The message would have to pass from `installesdtodmg.sh` -> `progresswatcher.py` -> `IEDWorkflow` -> `IEDCLIController`, and then a command-line option to enable it. If merely notifying via the log fits your use case, then that's ten minutes instead of half a day of work :)

**Comment by n8felton** [comment]
Personally, I have been renaming the file to be the build number that will be imaged onto a machine, eg `15G1108`. While it would be awesome if the app did this for me, asking the maintainer to spend ten minutes is easier than asking them to spend 4 hours on something they might not use themselves. Normally I would try and help with some code contributions, but I've yet to look at any source code for this project, so I'd probably spend more then 4 hours just trying to get familiar with the code flow.

As I currently only build an image once every month or two, me manually renaming the file is not a big deal.

**Comment by MagerValp** [comment]
I created a PR, which outputs the following to stdout and the log after building:

    Build success: OUTPUT_PATH='/Users/pelle/Downloads/t.dmg'
    Build success: OUTPUT_OSNAME='macOS'
    Build success: OUTPUT_OSVERSION='10.64.1'
    Build success: OUTPUT_OSBUILD='64X738'
    Build success: Notice: OS version changed from Mac OS X 9.0 53X248 to macOS 10.64.1 64X738

Try build [AutoDMG-1.7.1b484](https://www.dropbox.com/s/bjsqtknlgxw7g52/AutoDMG-1.7.1b484.dmg).

**Comment by MagerValp** [comment]
Shipped in 1.7.1.
